### PR TITLE
add option OrderCommandsByNameInHelpText

### PR DESCRIPTION
**Pull request by lvermeulen**
Fixes https://github.com/natemcmaster/CommandLineUtils/issues/147

### CODE REVIEW

**Comment by natemcmaster** [comment]
Hey @lvermeulen wanted to let you know I saw your PR. I'm should be able to review this closer in late September after I return from vacation.

Thanks,
-Nate

File: src/CommandLineUtils/CommandLineApplication.cs [hunk 1]
```diff
@@ -314,6 +314,11 @@ public CommandOption OptionHelp
         /// </summary>
         public TextWriter Error { get; set; }
 
+        /// <summary>
+        /// Determines if commands are ordered by name in generated help text
+        /// </summary>
+        public bool OrderCommandsByNameInHelpText { get; set; } = true;
```
[the remark this answers is not in the document]
Comment (lvermeulen) [reply src/CommandLineUtils/CommandLineApplication.cs]: I hear you! Let me make those changes.

File: src/CommandLineUtils/HelpText/DefaultHelpTextGenerator.cs [hunk 2]
```diff
@@ -23,6 +23,11 @@ public class DefaultHelpTextGenerator : IHelpTextGenerator
         /// </summary>
         protected DefaultHelpTextGenerator() { }
 
+        /// <summary>
+        /// Determines if commands are ordered by name in generated help text
+        /// </summary>
+        public bool OrderCommandsByNameInHelpText { get; set; } = true;
```
Comment (natemcmaster) [inline src/CommandLineUtils/HelpText/DefaultHelpTextGenerator.cs]: One last request: can we name this `SortCommandsByName`? Once that's in, I'll be happy to merge this PR.

Comment (lvermeulen) [reply src/CommandLineUtils/HelpText/DefaultHelpTextGenerator.cs]: Sure thing, coming up...

**Comment by natemcmaster** [comment]
Thanks @lvermeulen!

**Pull request merged**